planning/obstacle_avoidance_planner/src/node.cpp

```diff
@@ -1295,17 +1302,34 @@ void ObstacleAvoidancePlanner::insertZeroVelocityOutsideDrivableArea(
     const bool is_outside = drivable_area_utils::isOutsideDrivableAreaFromRectangleFootprint(
       traj_point, left_bound, right_bound, vehicle_param_, is_considering_footprint_edges_);
 
-    // only insert zero velocity to the first point outside drivable area
+    // only insert zero velocity to the first point outside drivable area (minus stop margin)
     if (is_outside) {
-      traj_points[i].longitudinal_velocity_mps = 0.0;
+      size_t stop_idx = i;
+      const auto & op_target_point = motion_utils::calcLongitudinalOffsetPoint(
+        traj_points, traj_point.pose.position, -1.0 * vehicle_stop_margin_outside_drivable_area_);
+
+      if (op_target_point) {
+        const auto target_point = op_target_point.get();
+        stop_idx = motion_utils::findNearestSegmentIndex(traj_points, target_point);
```

Comment:
I guess we should check if target point overlaps with nearest traj point or not. Otherwise even the margin is 0, nearest segment idx will nearest_point-1

```suggestion
        // confirm that target point doesn't overlap with the nearest trajectory point
        const size_t nearest_target_idx =
          motion_utils::findNearestIndex(traj_points, op_target_point.get());
        const auto dist = tier4_autoware_utils::calcDistance2d(
          traj_points.at(nearest_target_idx), op_target_point.get());
        const double overlap_threshold = 1e-3;

        if (dist > overlap_threshold) {
          const auto target_point = op_target_point.get();
          stop_idx = motion_utils::findNearestSegmentIndex(traj_points, target_point);
        }
```